src/lib.rs

```diff
@@ -105,6 +105,7 @@ macro_rules! register_module {
         // Mark this function as a global constructor (like C++).
         #[allow(improper_ctypes)]
         #[cfg_attr(target_os = "linux", link_section = ".ctors")]
+        #[cfg_attr(target_os = "android", link_section = ".ctors")]
```

Comment:
@dherman I'm not sure what other platforms this might impact. Would specifying `unix` match both `linux` and `android`? What about changing this to `all(not(windows), not(macos))`?

Reply:
TBH I don't really know where `.ctors` is "standardized" or how broad of a convention it is. It's an interesting question but in the interest of making progress, I'd say let's stick with the PR as written since it's the most conservative incremental improvement.

Reply:
I have another PR coming up to add iOS support and the only change in it would be one line of code here, to specify that the iOS case should look just like the macos (darwin) case, I guess because of LLVM.

Would you prefer that I update this PR with that, or should we do a separate PR for iOS?

Reply:
I'm fine either way—I'd say do whatever's easiest for you. Thank you for these contributions!